src/core/Codend.Application/Sprints/Commands/UpdateSprint/UpdateSprintCommand.cs

```diff
@@ -0,0 +1,82 @@
+using Codend.Application.Core;
+using Codend.Application.Core.Abstractions.Data;
+using Codend.Application.Core.Abstractions.Messaging.Commands;
+using Codend.Contracts.Requests;
+using Codend.Domain.Core.Extensions;
+using Codend.Domain.Entities;
+using Codend.Domain.Repositories;
+using Codend.Domain.ValueObjects;
+using FluentResults;
+using static Codend.Domain.Core.Errors.DomainErrors.General;
+
+namespace Codend.Application.Sprints.Commands.UpdateSprint;
+
+/// <summary>
+/// Command for updating properties in sprint with given id.
+/// </summary>
+/// <param name="SprintId">Id of the sprint that will be updated.</param>
+/// <param name="Name">New sprint name.</param>
+/// <param name="StartDate">New sprint start date.</param>
+/// <param name="EndDate">New sprint end date.</param>
+/// <param name="Goal">New sprint goal.</param>
+public record UpdateSprintCommand
+(
+    SprintId SprintId,
+    string? Name,
+    DateTime? StartDate,
+    DateTime? EndDate,
+    ShouldUpdateBinder<string?> Goal
+) : ICommand;
+
+/// <summary>
+/// <see cref="UpdateSprintCommand"/> handler.
+/// </summary>
+public class UpdateSprintCommandHandler : ICommandHandler<UpdateSprintCommand>
+{
+    private readonly IUnitOfWork _unitOfWork;
+    private readonly ISprintRepository _sprintRepository;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="UpdateSprintCommandHandler"/> class.
+    /// </summary>
+    public UpdateSprintCommandHandler(IUnitOfWork unitOfWork, ISprintRepository sprintRepository)
+    {
+        _unitOfWork = unitOfWork;
+        _sprintRepository = sprintRepository;
+    }
+
+    /// <inheritdoc />
+    public async Task<Result> Handle(UpdateSprintCommand request, CancellationToken cancellationToken)
+    {
+        var sprint = await _sprintRepository.GetByIdAsync(request.SprintId, cancellationToken);
+        if (sprint is null)
+        {
+            return DomainNotFound.Fail<Sprint>();
+        }
+
+        Result<SprintPeriod> periodResult = Result.Ok();
+        if (request.StartDate is not null && request.EndDate is not null)
+        {
+            periodResult = sprint.EditPeriod(
+                request.StartDate.Value.ToUniversalTime(),
```

Comment:
We have json converter that will convert datetime string into UTC datetime. So we don't need it here. We have to be sure to send proper utc (ISO 8601 standard) string to backend.

Reply:
So you telling me that we assume that we get valid data from the client. 🤔

Reply:
DateTime is stored with timezone and Postgres can't store timezones. If we don't call this method database will just refuse to store the date.

Reply:
Yes this ISO convention assures that the time is sent in utc standard. There are some libraries in js but normal Date() should work as well. Idk what are you talking about postgres not having timezones (timestampz) 

![image](https://github.com/CodendDev/Codend/assets/82521883/196c35d1-8a33-40fc-ac03-f9f5355248c1)

Nevertheless we should be aware of possible time problems.

Reply:
dobra nie zesraj sie

Reply:
💀️

Reply:
Postgres timestamp with time zone support only UTC.